docs/CaliptraIntegrationSpecification.md

```diff
@@ -725,6 +725,145 @@ The following code snippet and schematic diagram illustrate JTAG originating CDC
+| 2 | cptra_uc_rst_b | rvtop.veer.dec.tlu.exthaltff.genblock.dff.dout[7] | cptra_noncore_rst_b | cg.user_icg.clk_slcg_0_generated_icg.p_clkgate.vudpi0.q                   |
+| 3 | cptra_uc_rst_b | rvtop.veer.Gen_AXI_To_AHB.lsu_axi4_to_ahb*        | cptra_noncore_rst_b | entropy_src.u_entropy_src_core.u_caliptra_prim_packer_fifo_bypass.data_q* |
+| 4 | cptra_uc_rst_b | rvtop.veer.Gen_AXI_To_AHB.lsu_axi4_to_ahb*        | cptra_noncore_rst_b | entropy_src.u_entropy_src_core.u_caliptra_prim_packer_fifo_precon.data_q* |
+
+<br>
+
+**For violations in Sl No 1 and 2**, below in the schematic for the crossing. 
+
+![](./images/halt_status_rdc.png)
+
+This violation can be waived because if the CPU is halted, there is no way to trigger a firmware update reset as it is initiated by the microcontroller itself. Thus, it can be ensured that on *cptra_uc_rst_b*, *cpu_halt_status* will be at the reset value already, which in turn ensures that there are no glitches on the output clock of CG.
+
+**For violations in Sl No 3 and 4**,
+
+![](./images/entropy_RDC.png)
+
+Upon observing the above schematic, a RDC crossing violation is identified at the *data_q* flop (located on the right-hand side). The analysis for *prim_packer_fifo_bypass* is elaborated here, with the same scenario applying to *prim_packer_fifo_recon*.
+
+The fifo load and wr_data equations are as follows:-
+
+<span style="font-family: Consolas;">
+  assign pfifo_bypass_push = !es_bypass_mode ? 1'b0 :
+                             fw_ov_mode_entropy_insert ? fw_ov_fifo_wr_pulse :
+                             pfifo_postht_not_empty;
+  assign pfifo_bypass_wdata = fw_ov_mode_entropy_insert ? fw_ov_wr_data :
+                              pfifo_postht_rdata;
+</span>
+
+<br>
+Below are the various scenarios that can happen.
+
+| #Case | es_bypass_mode | fw_ov_mode_entropy_insert | pfifo_bypass_push | pfifo_bypass_wdata | Comment |
+| ------| -------------- | ------------------------- | ----------------- | ------------------ | ------- |
+| 1 | 0 | don't care | 0                   | don't care         | No RDC violations, since the recirculate mux on data_q (pink on lower right side) will prevent the flop to be loaded.                                           |
+| 2 | 1 | 0          | pfifo_postht_rdata  | pfifo_postht_rdata | No RDC violations, since recirculate mux will switch to the leg which is on the same reset domain (shown in purple on left side)                                |
+| 3 | 0 | 1          | fw_ov_fifo_wr_pulse | fw_ov_wr_data      | RDC violations will be suppressed since fw_ov_fifo_wr_pulse = 0 (red data lines which are on uc_rst will not be able to reach data_q since load_data will be 0) |
+
+Referring to the table above, if it can be ensured that *fw_ov_fifo_wr_pulse* is 0 when *cptra_uc_rst_b* is asserted, RDC crossings (Case #3) should be avoided. This condition is already addressed in the Constraints Section above.
```

Comment:
Minor updates: Referring to the preceding table, if it can be ensured that *fw_ov_fifo_wr_pulse* is 0 when *cptra_uc_rst_b* is asserted, RDC crossings (Case #3) can be avoided. This condition is described in the preceding Constraints section.

Reply:
thanks, updated